### PR TITLE
[OrderByComparator] LPD-26703 Clean up and Apply Singleton in layout-api, layout-page-template-api, layout-page-template-admin-web, layout-portlets-web  and message-boards-api module 

### PR DESCRIPTION
**Pull request by Valmir-unicap**
- Ticket: https://liferay.atlassian.net/browse/LPD-26703

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:stop

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:stop

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
That failure in SF is not related to my changes.

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
I did the `push` for fix conflicts

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
- That failure in `SF` isn't related to my changes. 
- I did the `push` for fix conflicts

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeCatchAllCheck.testjava [hunk 1]
```diff
@@ -181,13 +183,33 @@ public class UpgradeCatchAllCheck implements ConfigurationBeanDeclaration {
 		new KBTemplateModifiedDateComparator();
 		new KBTemplateTitleComparator();
 		new KBTemplateUserNameComparator();
+		new LayoutClassedModelUsageModifiedDateComparator();
 		new LayoutComparator();
+		new LayoutCreateDateComparator();
+		new LayoutModifiedDateComparator();
+		new LayoutPageTemplateEntryCreateDateComparator();
```
Comment (dantewang) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeCatchAllCheck.testjava:190]: @Valmir-unicap Please fix sorting, `LayoutPageTemplateEntry*` should come later than `LayoutPageTemplateCollection*`.

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava [hunk 2]
```diff
@@ -196,13 +198,33 @@ public class UpgradeCatchAllCheck implements ConfigurationBeanDeclaration {
 		KBTemplateModifiedDateComparator.getInstance(false);
 		KBTemplateTitleComparator.getInstance(false);
 		KBTemplateUserNameComparator.getInstance(false);
+		LayoutClassedModelUsageModifiedDateComparator.getInstance(true);
 		LayoutComparator.getInstance(false);
+		LayoutCreateDateComparator.getInstance(true);
+		LayoutModifiedDateComparator.getInstance(false);
+		LayoutPageTemplateEntryCreateDateComparator.getInstance(false);
```
Comment (dantewang) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava:205]: @Valmir-unicap The non-parametered constructors of `LayoutPageTemplateEntryCreateDateComparator` and `LayoutPageTemplateEntryModifiedDateComparator` pass `ascending = true` to the other constructor, so these 2 usages should be replaced by `getInstance(true)`.

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 3]
```diff
@@ -316,6 +316,318 @@
...
+		"from": "new LayoutPageTemplateCollectionNameComparator(boolean paramName)",
+		"issueKey": "LPD-26703",
+		"skipParametersValidation": true,
+		"to": "LayoutPageTemplateCollectionNameComparator.getInstance(param#0#)"
+	},
+	{
+		"from": "new LayoutPageTemplateEntryCreateDateComparator()",
+		"issueKey": "LPD-26703",
+		"skipParametersValidation": true,
+		"to": "LayoutPageTemplateEntryCreateDateComparator.getInstance(false)"
```
Comment (dantewang) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:455]: @Valmir-unicap Should be `true`

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 4]
```diff
@@ -316,6 +316,318 @@
...
+		"from": "new LayoutPageTemplateEntryCreateDateComparator(boolean paramName)",
+		"issueKey": "LPD-26703",
+		"skipParametersValidation": true,
+		"to": "LayoutPageTemplateEntryCreateDateComparator.getInstance(param#0#)"
+	},
+	{
+		"from": "new LayoutPageTemplateEntryModifiedDateComparator()",
+		"issueKey": "LPD-26703",
+		"skipParametersValidation": true,
+		"to": "LayoutPageTemplateEntryModifiedDateComparator.getInstance(false)"
```
Comment (dantewang) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:467]: @Valmir-unicap should be `true`

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@Valmir-unicap Please resolve conflict and fix the sorting and mapping in the replacements, thanks!

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
That failure in `SF` isn't related to my changes.
I did the `push` for fix conflicts

**Comment by dantewang** [comment]
ci:forward:force

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
@Valmir-unicap Plz resolve conflict and then forward, thanks!

**Comment by Valmir-unicap** [comment]
ci:report:58674572

**Comment by Valmir-unicap** [comment]
Thanks, @dantewang! :)

**Comment by Valmir-unicap** [comment]
ci:forward:force

**Comment by Valmir-unicap** [comment]
ci:report:59111324

**Comment by Valmir-unicap** [comment]
ci:report:59111324

**Comment by Valmir-unicap** [comment]
ci:forward:force

**Pull request closed without merging**